arch/arm/soc/st_stm32/stm32l4/soc.c

```diff
@@ -31,6 +31,12 @@ static int stm32l4_init(struct device *arg)
 
 	key = irq_lock();
 
+	/* Enable flash instruction prefetch, instruction cache memory,
+	 * and data cache memory. */
+	LL_FLASH_EnablePrefetch();
+	LL_FLASH_EnableInstCache();
+	LL_FLASH_EnableDataCache();
```

Comment:
According to the reference manual instruction cache memory and data cache memory are enabled after system reset.

Reply:
Hmm you are correct, so the improvement I measured where only due to the prefetch, impressive.

Closing this pull request, I'll reopen one with prefetch only and some numbers.